stumpy/snippets.py

```diff
@@ -10,6 +10,30 @@
 from .aampdist_snippets import aampdist_snippets
 
 
+def _get_mask_slices(idx, mask):
+    """
+    For a boolean vector mask, returns the slices of indices at which the mask is True.
+
+    Parameters
+    ----------
+    mask: boolean ndarray
+
+    Returns
+    -------
+    slices: slices of indices where the mask is True
+
+    """
+    m1 = np.r_[0, mask]
+    m2 = np.r_[mask, 0]
+
+    (starts, ) = np.where(~m1 & m2)
+    (ends, ) = np.where(m1 & ~m2)
+
+    slices = np.c_[[idx] * len(starts), starts, ends]
```

Comment:
I think this function with `idx` is too specialized/specific. I think it is better to stick with the more generic (and reusuable):

```
def _get_mask_slices(mask):
    m1 = np.r_[0, mask]
    m2 = np.r_[mask, 0]
    starts, = np.where(~m1 & m2)
    ends, = np.where(m1 & ~m2)
    return np.c_[starts, ends]
```

And then post-process the array once it is returned. Otherwise, it isn't clear what `idx` is doing